File: frontend/lib/models/dose_log_model.dart
import 'package:flutter/material.dart';

enum DoseStatus { upcoming, taken, missed, skipped }

class DoseLog {
  final String id;
  final String medicineId;
  final String medicineName;
  final String dosage;
  final DateTime scheduledTime;
  DoseStatus status;
  DateTime? actionTime;
  final Color color;

  DoseLog({
    required this.id,
    required this.medicineId,
    required this.medicineName,
    required this.dosage,
    required this.scheduledTime,
    this.status = DoseStatus.upcoming,
    this.actionTime,
    required this.color,
  });
}

class NotificationPrefs {
  bool soundEnabled;
  bool vibrationEnabled;
  int snoozeMinutes;
  bool missedDoseAlerts;
  String soundName;

  NotificationPrefs({
    this.soundEnabled = true,
    this.vibrationEnabled = true,
    this.snoozeMinutes = 10,
    this.missedDoseAlerts = true,
    this.soundName = 'Chime',
  });
}

File: backend/src/models/dose_log.model.js
const { DataTypes } = require("sequelize");
const { sequelize } = require("../config/db");

const DoseLog = sequelize.define(
  "DoseLog",
  {
    id: {
      type: DataTypes.BIGINT.UNSIGNED,
      autoIncrement: true,
      primaryKey: true,
    },
    userId: {
      type: DataTypes.BIGINT.UNSIGNED,
      allowNull: false,
      references: {
        model: "users",
        key: "id",
      },
    },
    medicineId: {
      type: DataTypes.BIGINT.UNSIGNED,
      allowNull: false,
      references: {
        model: "medicines",
        key: "id",
      },
    },
    medicineName: {
      type: DataTypes.STRING(150),
      allowNull: false,
    },
    dosage: {
      type: DataTypes.STRING(50),
      allowNull: false,
    },
    scheduledTime: {
      type: DataTypes.DATE,
      allowNull: false,
    },
    status: {
      type: DataTypes.ENUM("upcoming", "taken", "missed", "skipped"),
      allowNull: false,
      defaultValue: "upcoming",
    },
    actionTime: {
      type: DataTypes.DATE,
      allowNull: true,
    },
    color: {
      type: DataTypes.STRING(7),
      allowNull: false,
      defaultValue: "#0EA5A0",
    },
  },
  {
    tableName: "dose_logs",
    timestamps: true,
    createdAt: "createdAt",
    updatedAt: "updatedAt",
  }
);

module.exports = DoseLog;